File: OC-Swift-Bridge/OC-Swift-Bridge/OC/ViewController.m
//
//  ViewController.m
//  OC-Swift-Bridge
//
//  Created by lzh on 2021/5/21.
//

#import "ViewController.h"
#import "HelloController.h"
// MARK: 引入头文件，可以调用模块内的swift内容
// 格式: #import <TargetName-Swift.h>
#import <OC_Swift_Bridge-Swift.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    UIButton *button = [[UIButton alloc] initWithFrame:CGRectMake(0, 0, 100, 100)];
    button.backgroundColor = UIColor.blackColor;
    [self.view addSubview:button];
    [button addTarget:self action:@selector(tapButton) forControlEvents:UIControlEventTouchUpInside];
}

- (void)tapButton {
//    [self presentViewController:[[HelloController alloc] init] animated:YES completion:nil];
    [self presentViewController:[[NewHelloController alloc] init] animated:YES completion:nil];
}


@end

File: SVG/SVG/ViewController.swift
//
//  ViewController.swift
//  SVG
//
//  Created by lzh on 2021/6/30.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        view.backgroundColor = .white

        view.addSubview(localButton)
        view.addSubview(networkButton)
    }
    
    lazy var localButton: UIButton = {
        let button = UIButton(frame: .init(x: 20, y: 100, width: 100, height: 50))
        button.backgroundColor = .black
        button.setTitle("本地SVG", for: .normal)
        button.addTarget(self, action: #selector(openLocal), for: .touchUpInside)
        return button
    }()
    
    @objc func openLocal() {
        navigationController?.pushViewController(LocalViewController(), animated: true)
    }
    
    lazy var networkButton: UIButton = {
        let button = UIButton(frame: .init(x: 220, y: 100, width: 100, height: 50))
        button.backgroundColor = .black
        button.setTitle("网络SVG", for: .normal)
        button.addTarget(self, action: #selector(openNetwork), for: .touchUpInside)
        return button
    }()
    
    @objc func openNetwork() {
        navigationController?.pushViewController(NetworkViewController(), animated: true)
    }
}

